### PR TITLE
Add partial fit to SpacyClassifier

### PR DESCRIPTION
**Pull request by nsorros**
Adds partial fit to spacy classifier which allows training to take place for datasets that do not fit in memory. Spacy by default deals with this quite nicely so we are just translating it to sklearn convention.

### CODE REVIEW

File: tests/test_spacy_classifier.py [hunk 1]
```diff
@@ -45,3 +45,25 @@ def test_multilabel_Y_list():
...
+        [1,0,0,0],
+        [0,0,1,1],
+        [0,1,0,0],
+        [0,1,1,0]
+    ]
+
+    model = SpacyClassifier()
+    for x, y in zip(X, Y):
+        model.partial_fit([x], [y])
+    assert model.score(X, Y) > 0.4
```
Comment (aCampello) [inline tests/test_spacy_classifier.py]: We could lower this to 0.3, or even just `model.score(X, Y) > 0` to ensure the pipeline passes? I think this is trying to test whether the code runs correctly and output a non-zero number, which probably means that it is training.

File: wellcomeml/ml/spacy_classifier.py [hunk 2]
```diff
@@ -132,6 +132,42 @@ def fit(self, X, Y):
...
+        """
+        Args:
+            X: 1d list of documents
+            Y: 2d numpy array (nb_examples, nb_labels)
+        TODO: Generalise to y being 1d
+        """
+        Y = np.array(Y)
+
+        if not hasattr(self, 'unique_labels'):
+            # We could also use classes here
```
Comment (aCampello) [inline wellcomeml/ml/spacy_classifier.py]: Yes. Either add a class check, or remove the parameter from above? It seems it is not being used...

Comment (nsorros) [reply wellcomeml/ml/spacy_classifier.py]: The parameter is needed to support the sklearn API

File: tests/test_spacy_classifier.py (hunk 1 above)
Comment (nsorros) [reply tests/test_spacy_classifier.py]: I think we need to make sure it learns something better than chance. Reducing to 0.3 for now.

**Pull request merged**